mllib/src/test/scala/org/apache/spark/mllib/clustering/LDASuite.scala

```diff
@@ -99,9 +99,15 @@ class LDASuite extends SparkFunSuite with MLlibTestSparkContext {
 
     // Check: per-doc topic distributions
     val topicDistributions = model.topicDistributions.collect()
+
     //  Ensure all documents are covered.
-    assert(topicDistributions.length === tinyCorpus.length)
-    assert(tinyCorpus.map(_._1).toSet === topicDistributions.map(_._1).toSet)
+    // SPARK-5562. since the topicDistribution returns the distribution of docsI over topics .
+    // for empty docs, since the distribution of topic won't sum to 1 (and hence it is not a pdf)
+    // So the output will not contain the empty docs and hence we modify the unittest to
+    // compare against nonEmptyTinyCorpus
+    val nonEmptyTinyCorpus = getNonEmptyDoc(tinyCorpus)
+    assert(topicDistributions.length === nonEmptyTinyCorpus.length)
```

Comment:
As you had pointed out later. going with the options 1 and updating the docs at mllib-clustering.md would be better and I agree with you. I will pull the the changes in.